freqtrade/optimize/hyperopt.py

```diff
@@ -367,7 +367,7 @@ def _get_results_dict(self, backtesting_results, min_date, max_date,
         }
 
     def get_optimizer(self, dimensions: List[Dimension], cpu_count) -> Optimizer:
-        estimator = self.custom_hyperopt.generate_estimator()
+        estimator = self.custom_hyperopt.generate_estimator(dimensions)
```

Comment:
I'm not sure about implementing it this way - it's a change breaking all strategies that implement this function.

a potentially better way might be to simply assign dimensions to custom_hyperopt. there's little to no difference if it's used as `self.dimensions` or `self.custom_hyperopt.dimensions` - and you could access it this way:

``` python
class MyAwesomeStrategy(IStrategy):
    class HyperOpt:
        def generate_estimator(self)
            ... = len(self.dimensions)
```

Reply:
Yes, I'm sure there's a better way to accomplish this. Would you mind being a bit more specific about your idea? Are you suggesting to initialize `self.dimensions` [here](https://github.com/italodamato/freqtrade-1/blob/a4dbdb549d772b488305cb12727c5a8375b0cb2d/freqtrade/optimize/hyperopt_interface.py#L36)?

Reply:
no not really
if you look at the `hyperopt.py` file - there's only 4(ish) usages of `self.dimension`.

except for the first (which needs to be moved down a bit within that function), these can be changed to `self.custom_hyperopt.dimension` - so you'll have it automatically available.

`get_optimizer()` also don't "really" needs it as argument (it's available as class-attribute anyway, so that's something that's been historically this way) - and it'll be passed by reference anyway

Reply:
hmm, the suggested approach above does not work as we never create an instance of the HyperOpt childclass, therefore i guess your approach is correct 👍 

the only thing:
would you mind calling the function with keyword args (basically `(dimensions=dimensions)` - as well as updating the docs for this (basically search for `generate_estimator`).

I'd want to add a `**kwargs` argument to the function definition as well - so we can easily add new parameters without breaking existing strategies should that be necessary in the future.